### ISSUE TITLE
Probably a bug on hyperparameter research

### ISSUE THREAD

**Issue by gokhantahil** [deepchem/deepchem#2502]
Hello,

I have a problem with the hyperparameter research on Deepchem.

After the hyperparameter research, the program returns me a RF model as following :

SklearnModel(model=RandomForestRegressor(criterion='mae', max_depth=8,
                                         min_weight_fraction_leaf=0,
                                         n_estimators=120))

with the R2 of Testing : 64.93

Then I tried to compare the original RF result with it so I created a model as following :

SklearnModel(model=RandomForestRegressor(criterion='mae', max_depth=8,
                                         min_samples_leaf=3,
                                         min_samples_split=3,
                                         min_weight_fraction_leaf=0,
                                         n_estimators=120))

It gives me R2 of Testing : 72.38 and validation 68.32

Then in the next test, I added one more hyperparameter : 

'min_samples_leaf': [1,3]

So what i was expecting that for min_samples_leaf : 3 I should have the same R2 score for validation.

The program returns the validation scores as following :

_criterionmae_max_depth_8_min_samples_leaf_1_min_samples_split_3_min_weight_fraction_leaf_0_n_estimators_120_n_jobs_-1: 0.6514039887382881
_criterionmae_max_depth_8_min_samples_leaf_3_min_samples_split_3_min_weight_fraction_leaf_0_n_estimators_120_n_jobs_-1: 0.6667454007419835

But it decreased. It choses the yellow one as best hyperparameters since it had better validation score.

Since it's the same hyperparameter that we used in the previous model, I was expecting the same R2 of Testing as well but it gives me 75.14.

I used a seed, nothing has been changed between these tests the training, validation and testing dataset didn't change, no idea why it happens. Here is the last part of the script :

```
def rf_model_builder(model_dir, **model_params):
  sklearn_model = RandomForestRegressor(**model_params)
  return dc.models.SklearnModel(sklearn_model, model_dir)
params_dict = {
    # 'n_estimators': [i * 5 for i in range(22, 27)],
    # 'max_depth': [6, 7, 8],
    # 'criterion': ['mse', 'mae'],
    # 'min_samples_split': [0.9, 2, 3, 4],
    # 'min_samples_leaf': [1, 2, 3, 4],
    # "min_weight_fraction_leaf": [0, 0.1, 0.2, 0.3],
'n_estimators': [120],
    'max_depth': [8],
    'criterion': ['mae'],
    'min_samples_split': [3],
    'min_samples_leaf': [1,3],
    "min_weight_fraction_leaf": [0],
    'n_jobs':[-1]
}

metric = dc.metrics.Metric(dc.metrics.r2_score)
optimizer = dc.hyper.GridHyperparamOpt(rf_model_builder)
best_rf, best_rf_hyperparams, all_rf_results = optimizer.hyperparam_search(
    params_dict, train_dataset, valid_dataset,
    metric=metric, use_max=True)
for key, value in all_rf_results.items():
    print(f'{key}: {value}')
print('Best hyperparams:', best_rf_hyperparams)

print(best_rf)

task = "DeltaG"
predicted_test = best_rf.predict(test_dataset)
print(predicted_test)
true_test = test_dataset.y
print((r2_score(true_test, predicted_test))

```

Could you please tell me if there is a mistake? Probably it's a bug.

Thanks

**Comment by gokhantahil** [comment]
The problem has been solved with random_state value in RandomForestRegressor.

`sklearn_model = RandomForestRegressor(**model_params,random_state=42)`
